docs/source/pages/lightning.rst

```diff
@@ -118,85 +96,125 @@ If ``on_epoch`` is True, the logger automatically logs the end of epoch metric v
+mixed as it can lead to wrong results.
+
 .. note::
 
-    The ``.reset()`` method of the metric will automatically be called at the end of an epoch within lightning only if you pass
-    the metric instance inside `self.log <https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html#logging-from-a-lightningmodule>`_.
-    Also if you are calling ``.compute`` by yourself, you need to call the ``.reset()`` too.
+    When using any Modular metric, calling ``self.metric(...)`` or ``self.metric.forward(...)`` serves the dual purpose of calling ``self.metric.update()``
+    on its input and simultaneously returning the metric value over the provided input. So if you are logging a metric *only* on epoch-level (as in the
+    example above), it is recommended to call ``self.metric.update()`` directly to avoid the extra computation.
 
     .. testcode:: python
 
         class MyModule(LightningModule):
 
             def __init__(self):
                 ...
-                self.train_acc = torchmetrics.Accuracy()
-                self.train_precision = torchmetrics.Precision()
+                self.valid_acc = torchmetrics.Accuracy()
 
-            def training_step(self, batch, batch_idx):
-                x, y = batch
-                preds = self(x)
+            def validation_step(self, batch, batch_idx):
+                logits = self(x)
                 ...
+                self.valid_acc.update(logits, y)
+                self.log('valid_acc', self.valid_acc, on_step=True, on_epoch=True)
 
-                # this will reset the metric automatically at the epoch end
-                self.train_acc(preds, y)
-                self.log('train_acc', self.train_acc, on_step=True, on_epoch=False)
 
-                # this will not reset the metric automatically at the epoch end
-                precision = self.train_precision(preds, y)
-                self.log('train_precision', precision, on_step=True, on_epoch=False)
+***************
+Common pitfalls
+***************
 
-            def training_epoch_end(self, outputs):
-                # this will compute and reset the metric automatically at the epoch end
-                self.log('train_epoch_accuracy', self.training_acc)
+The following contains a list of pitfalls to be aware of:
 
-                # this will not reset the metric automatically at the epoch end so you
-                # need to call it yourself
-                mean_precision = self.precision.compute()
-                self.log('train_epoch_precision', mean_precision)
-                self.precision.reset()
+* If using metrics in data parallel mode (dp), the metric update/logging should be done
+  in the ``<mode>_step_end`` method (where ``<mode>`` is either ``training``, ``validation``
+  or ``test``). This is due to metric states else being destroyed after each forward pass,
+  leading to wrong accumulation. In practice do the following:
```

Comment:
```suggestion
  or ``test``). This is because ``dp`` split the batches during the forward pass and metric states are destroyed after each forward pass, thus leading to wrong accumulation. In practice do the following:
```